components/automate-ui/src/app/pages/create-notification-modal/create-notification-modal.component.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { EventEmitter } from '@angular/core';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { ReactiveFormsModule, FormBuilder } from '@angular/forms';
+import { MatSelectModule } from '@angular/material/select';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { MockComponent } from 'ng2-mock-component';
+
+import { CreateNotificationModalComponent } from './create-notification-modal.component';
+
+describe('CreateNotificationModalComponent', () => {
+  let component: CreateNotificationModalComponent;
+  let fixture: ComponentFixture<CreateNotificationModalComponent>;
+
+  beforeEach(async () => {
+    TestBed.configureTestingModule({
+      declarations: [
+        MockComponent({ selector: 'chef-button', inputs: ['disabled'] }),
+        MockComponent({ selector: 'chef-error' }),
+        MockComponent({ selector: 'chef-form-field' }),
+        MockComponent({ selector: 'chef-heading' }),
+        MockComponent({ selector: 'chef-icon' }),
+        MockComponent({ selector: 'chef-loading-spinner' }),
+        MockComponent({ selector: 'chef-page-header' }),
+        MockComponent({ selector: 'chef-subheading' }),
+        MockComponent({ selector: 'chef-toolbar' }),
+        MockComponent({ selector: 'chef-table' }),
+        MockComponent({ selector: 'chef-thead' }),
+        MockComponent({ selector: 'chef-tbody' }),
+        MockComponent({ selector: 'chef-tr' }),
+        MockComponent({ selector: 'chef-th' }),
+        MockComponent({ selector: 'chef-td' }),
+        MockComponent({ selector: 'a', inputs: ['routerLink'] }),
+        MockComponent({ selector: 'chef-modal',
+          inputs: ['visible'],
+          outputs: ['close']
+        }),
+        CreateNotificationModalComponent
+      ],
+      imports: [
+        MatSelectModule,
+        ReactiveFormsModule
+      ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateNotificationModalComponent);
+    component = fixture.componentInstance;
+    component.createForm = new FormBuilder().group({
+      id: ['', null],
+      name: ['', null],
+      url: ['', null],
+      targetType: ['', null],
+      ruleType: ['', null],
+      targetUrl: ['', null],
+      username: ['', null],
+      password: ['', null]
+    });
+    component.conflictErrorEvent = new EventEmitter();
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```

Comment:
Please add tests of the component's code.
(If this is replacing notification-form.component.spec.ts, are any of those tests able to be migrated over here?)

Reply:
I have have covered in notification.component.ts